geonode_mapstore_client/client/js/epics/index.js

```diff
@@ -33,36 +32,25 @@ import get from 'lodash/get';
  */
 import { showCoordinateEditorSelector } from '@mapstore/framework/selectors/controls';
 
-/**
- * When a user selects a layer, the app checks for layer editing permission.
- */
-export const _setFeatureEditPermission = (action$, { getState } = {}) =>
-    action$.ofType(SELECT_NODE).filter(({ nodeType }) => nodeType === "layer" && !getConfigProp("disableCheckEditPermissions"))
-        .switchMap(() => {
-            const layer = getSelectedLayer(getState() || {});
-            return layer ? layerEditPermissions(layer)
-                .map(permissions => setPermission(permissions))
-                .startWith(setPermission({ canEdit: false }))
-                .catch(() => Rx.Observable.empty()) : Rx.Observable.of(setPermission({ canEdit: false }));
-        });
-/**
- * When a user selects a layer, the app checks for style editing permission.
- * INIT_STYLE_SERVICE si needed for map editing, it ensures an user has permission to edit style of a specific layer retrieved from catalog
- */
-export const _setStyleEditorPermission = (action$, { getState } = {}) =>
-    action$.ofType(INIT_STYLE_SERVICE, SELECT_NODE)
-        .filter(({ nodeType }) =>
-            nodeType && nodeType === "layer" && !getConfigProp("disableCheckEditPermissions")
-            || !nodeType && !getConfigProp("disableCheckEditPermissions"))
-        .switchMap((action) => {
-            const layer = getSelectedLayer(getState() || {});
-            return layer
-                ? styleEditPermissions(layer)
-                    .map(({ canEdit }) => setEditPermissionStyleEditor(canEdit))
-                    .startWith(setEditPermissionStyleEditor(action.canEdit))
-                    .catch(() => Rx.Observable.empty())
-                : Rx.Observable.of(setEditPermissionStyleEditor(false));
-        });
+
+export const checkFeatureAndEditorPermissions = (action$, { getState } = {}) =>
```

Comment:
All the epics are merged in a single object including the one form mapstore so their name should be unique. This is a particular epic used to update layer permission so I would change the name to: `gnCheckSelectedLayerPermissions`. (in this particular case we could also add the gn prefix). Replace the name also in other files